### ISSUE TITLE
Move disk image to SMB Share or other network storage

### ISSUE THREAD

**Issue by Simbaclaws** [utmapp/UTM#5417]
Hi there,

I'm trying to move one of my Mac OS hypervisor UTM disk images from the local Mac OS machine on a m2 Mac mini, to my TrueNAS Scale SMB Share, where I have more storage available.

I tried using the move option within the UTM app, but whenever I do this, it just keeps on loading and doesn't do anything. I let it go for almost 5 days on end without anything happening on the SMB share or on the local machine.

What can I do to move the disk image file with the utm file from my local machine to a network storage share such as SMB, NFS or iSCSI? I'd preferably use SMB.

This will allow people to have a NAS storage device that can be used to store disk images for UTM, so you can have way more storage for a multitude of virtual machines on the Mac.

I'm not even sure if booting a virtual machine from network storage would cause lots of performance issues or what not, but in any case, I need this for a use case that I have.

Please help me figure this out.

I've already tried moving the utm file from:
```/Users/myuser/Library/Containers/com.utmapp.UTM/Data/Documents/my-virtual-machine.utm```
to a network SMB share.

But when I import this utm file from the smb share in utm, it will fail to start the virtual machine.

Is this something that can be implemented in the future perhaps? As a feature?

Hopefully someone can help me out here.

Yours sincerely,
Hylke

**Comment by Teressss** [comment]
yeah, same problem here. It would be so great to run VMs from disk image stored on NAS :(

**Comment by mgrzybek** [comment]
Hi,

Same problem here using my Apple M1 using AFP or SMB shares:
- The QEMU engine does not want to use the disks.
- The native Apple virtualization makes the disks read-only.

**Comment by antoninsiska** [comment]
I have the same problems, I recently tried to store the UTM file on NAS and when I tied to run it, error comes. I also have MacBook M1 and Synology NAS

**Comment by jpg0** [comment]
FYI it is possible to get this working using remote block storage, instead of a remote FS. I tried using NBD (OSX client, linux server) and it worked if passing this via params to qemu (-drive file=nbd://server/image.raw). It's awkward, has no UI and no validation or decent error messages, so I'd much prefer SMB.
